data/language.go

```diff
@@ -0,0 +1,61 @@
+package data
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-enry/go-enry/v2"
+)
+
+func determineMostCommonLangugage(languageCount map[string]int) string {
+	max := 0
+	mostCommonLang := ""
+
+	for key, value := range languageCount {
+		if max < value {
+			max = value
+			mostCommonLang = key
+		}
+	}
+
+	return mostCommonLang
+}
+
+// InferLanguage detects the programming used in the provided work directory .
+func inferLanguage(workDir string) (string, error) {
+	languageCount := map[string]int{}
```

Comment:
instead of doing it in the whole directory, I think we should infer language in the work directory for each effx.yaml. what if we have a mix of services in a directory with different languages? like a typescript service and a go service, but most of the services are go? i dont want my typescript service to say `language:go`

Reply:
that's actually how this works btw :)

this function get's called for every directory that contains each effx.yaml

Reply:
maybe `workDir` is a misleading variable name.

Reply:
ohh ok.